lib/lightning/invocation.ex

```diff
@@ -765,6 +834,46 @@ defmodule Lightning.Invocation do
     |> Repo.transaction()
   end
 
+  # Query dataclips for cron-triggered jobs, including the next run state
+  defp list_dataclips_with_cron_state(job_id, user_filters, opts) do
+    # Get the next cron run dataclip (always needed for the ID)
+    next_cron_dataclip = get_next_cron_run_dataclip(job_id, dynamic(true))
+    next_cron_run_dataclip_id = next_cron_dataclip && next_cron_dataclip.id
+
+    # Check if next cron dataclip matches user filters
+    include_next_cron? =
+      next_cron_dataclip &&
+        dataclip_matches_filters?(next_cron_dataclip, user_filters)
+
+    # Get regular dataclips, excluding next cron if it will be included to avoid duplication
+    filters =
+      if include_next_cron?,
+        do: Map.put(user_filters, :exclude_id, next_cron_dataclip.id),
+        else: user_filters
+
+    regular_dataclips = list_dataclips_for_job(%Job{id: job_id}, filters, opts)
+
+    # Combine results with next cron dataclip first if it matches filters
+    dataclips =
+      if include_next_cron?,
+        do: [next_cron_dataclip | regular_dataclips],
+        else: regular_dataclips
+
+    {dataclips, next_cron_run_dataclip_id}
+  end
+
+  # Check if a dataclip matches the user filters (applied in Elixir)
+  defp dataclip_matches_filters?(dataclip, user_filters) do
```

Comment:
Hey @taylordowns2000 , I'm planning on getting rid of this in favor of reusing the query filter. 
The downside is we will potentially make 1 more trip to the db but I think it's fine compared to maintaining the filter in 2 different places